server/routes/slack/slack_events.py

```
@@ -556,4 +589,207 @@ def _handle_suggestion_details(payload: dict, action: dict, slack_user_id: str,
+
+                # 4. AUTHORIZE: RLS already scopes reads to the clicker's org;
+                #    this exists to give the right message instead of a
+                #    confusing "not found", and as defence in depth.
+                cursor.execute(
+                    "SELECT org_id, status FROM hpa_vpa_recommendations WHERE id = %s::uuid",
+                    (rec_id,),
+                )
+                row = cursor.fetchone()
+                if not row or row[0] != clicker_org_id:
+                    logger.warning(
+                        f"Slack user {sanitize(slack_user_id)} tried to dismiss recommendation outside their org"
```

Ruff (0.16.0)

[warning] Logging statement uses f-string

(G004)